daemons/controld/controld_remote_ra.c

```diff
@@ -182,16 +182,16 @@ remote_node_up(const char *node_name)
     CRM_CHECK(node_name != NULL, return);
     crm_info("Announcing pacemaker_remote node %s", node_name);
 
-    /* Clear node's operation history. The node's transient attributes should
-     * and normally will be cleared when the node leaves, but since remote node
-     * state has a number of corner cases, clear them here as well, to be sure.
-     */
+    // Clear node's operation history.
     call_opt = crmd_cib_smart_opt();
     erase_status_tag(node_name, XML_CIB_TAG_LRM, call_opt);
-    erase_status_tag(node_name, XML_TAG_TRANSIENT_NODEATTRS, call_opt);
 
-    /* Clear node's probed attribute */
-    update_attrd(node_name, CRM_OP_PROBED, NULL, NULL, TRUE);
+    /* The node's transient attributes should and normally will be cleared when
+     * the node leaves, but since remote node state has a number of corner
+     * cases, clear them here as well, to be sure. Importantly, this includes
+     * the CRM_OP_PROBED ("probe_complete") attribute.
+     */
+    update_attrd_clear_node(node_name, TRUE);
 
```

Comment:
update_attrd(node_name, CRM_OP_PROBED, NULL, NULL, TRUE);

Reply:
Looking at the code again, I am a little confused. I believe the CRM_OP_PROBED attribute should be unnecessary since 8f76b782 . Also, remote_node_up() explictly clears it in the case of a remote node. I'm not completely sure, but I think we can remove all references to CRM_OP_PROBED from the controller (we probably need to keep the references in the scheduler for regression test purposes).

Reply:
Hi Ken,

I understood this cause.

When attrd now handles remote node, we use this "probe_complete" attribute to register remote node.(crm_remote_peer_get())
Therefore, unless the "probe_complete" attribute is sent from crmd, attrd will not recognize the remote node properly.

```
(snip)
static attribute_value_t *
attrd_lookup_or_create_value(GHashTable *values, const char *host, xmlNode *xml)
{
    attribute_value_t *v = g_hash_table_lookup(values, host);
    int is_remote = 0;

    crm_element_value_int(xml, F_ATTRD_IS_REMOTE, &is_remote);
    if (is_remote) {
        /* If we previously assumed this node was an unseen cluster node,
         * remove its entry from the cluster peer cache.
         */
        crm_node_t *dup = crm_find_peer(0, host);

        if (dup && (dup->uuid == NULL)) {
            reap_crm_member(0, host);
        }

        /* Ensure this host is in the remote peer cache */
        CRM_ASSERT(crm_remote_peer_get(host) != NULL);
    }
(snip)
```
By the way, the attribute at this time works properly even if it is not "probe_complete".

```
#if 0
//YAMAUCHI
    /* Set the CRM_OP_PROBED ("probe_complete") attribute. */
    update_attrd(node_name, CRM_OP_PROBED, NULL, NULL, TRUE);
#endif
    update_attrd(node_name, "remote_up", NULL, NULL, TRUE);
```

It is like a magic trick to be recognized by the remote node to the last.

It may be better to fix it in the future, but this update_attrd () is necessary at present.

Best Regards,
Hideo Yamauchi.

Reply:
> It is like a magic trick to be recognized by the remote node to the last.

Ah, I see, thanks.
 
> It may be better to fix it in the future, but this update_attrd () is necessary at present.

Agreed.